### PR TITLE
Data table linking-related updates

### PR DESCRIPTION
**Pull request by katherinebernstein**
- update SetOriginIdToNonNullUpgrade to add a new origin_id column with default value (faster)
- split out data primary key change to separate upgrade

### CODE REVIEW

**Comment by katherinebernstein** [comment]
@you-leee The origin_id column renames will affect materialized views, since they will still be pointing at the old column after this runs. I think as long as we mark them all as needing refreshing, they'll get updated to use the new column, but I wanted to double check that understanding first

**Comment by you-leee** [comment]
I'm not sure, since the created views in openlattice database (`Assembler.createOrUpdateProductionViewOfEntitySet`) we import to organization databases and then use to create materialized views, won't be re-created. Plus now, if we mark materialized entity sets as "needing refresh" (`OrganizationEntitySetFlag.DATA_UNSYNCHRONIZED`) will only execute `REFRESH MATERIALIZED VIEW`

**Comment by you-leee** [comment]
So, yeah, just tried it locally too. The view in openlattice database needs to be re-created in order to ensure, that in materialized views we use the new origin_id. Basically for every materialized view, we need to re-create those views and then re-materialize too.

File: src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    override fun getSupportedVersion(): Long {
+        return Version.V2019_10_03.value
+    }
+
+    override fun upgrade(): Boolean {
+
+        val pkey = (PostgresDataTables.buildDataTableDefinition().primaryKey + ORIGIN_ID).joinToString(",") { it.name }
+
+        val dropPkey = "ALTER TABLE ${DATA.name} DROP CONSTRAINT ${DATA.name}_pkey"
+        val updatePkey = "ALTER TABLE ${DATA.name} ADD PRIMARY KEY ($pkey)"
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt]: I would also rename the index corresponding to the primary key and delete it *after* the new index has been created.

**Review by geekbeast — CHANGES_REQUESTED** [review]
suggestions on handling index roll over without interruption

File: src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt (hunk 1 above)
Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt]: I don't think postgres will allow you to have multiple primary keys at once

Comment (katherinebernstein) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt]: although if there's a way to rename the constraint and remove its primary key status then that would be ideal

**Review by makosblade — APPROVED** [review]

File: src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt (hunk 1 above)
Comment (geekbeast) [reply src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt]: Primary key is a constraint, the index is separate from the constraint.

**Review by geekbeast — APPROVED** [review]

**Pull request merged**